src/modules/commands.py

```diff
@@ -32,44 +32,72 @@ def __init__(self, commandLine):
     
     def commandBody(self, variables):
         """
-        Promts user for Postcode
-        Searches Postcode and prints coords
-        Prints erors if multiple values found or Postcode is not found
+        Prompts user for a postcode and finds centre coordinate of that postcode
+        Prompts user for a radius and filters crime data within that radius of postcode centre
+        Prompts user for method of sorting data
+        Prompts user for file name for CSV
+        Results are outputted into CSV with specified file name
+        Uses are re-prompted if invalid information is inputted
         """
-
-        input = self.prompt("Please enter a Postcode")
-        result = find_postcode_coordinate(input, variables['postcodes'])
-        if result == -1:
-            print("Could not find postcode")
-        elif result == -2:
-            print("Multiple postcodes found")
-        else:
-            print("Latitude: " + result[0] + " Longitude: " + result[1])
-            lat = result[0]
-            long = result[1]
-            postcodelatlong = []
-            postcodelatlong.append(float(lat))
-            postcodelatlong.append(float(long))
-            input = self.prompt("Please enter a search radius in km")
-            radius = int(input)
-            filtered_data = filterData(variables['crimedata'], postcodelatlong, radius)
-            input = self.prompt("How would you like the data sorted? By crime category, date (recent first) or distance?" )
-            
-            if input == "crime category":
-                key = "Crime type"
-                reverse = False
-            elif input == "date":
-                key = "Month"
-                reverse = True
-            elif input == "distance":
-                key = "Distance"
-                reverse = False
-            
-            sorted_data = listOfDictSort(filtered_data, key, reverse, dateFormat="")
+        while True:
+            try:
+                input = self.prompt("Please enter a Postcode")
+                result = find_postcode_coordinate(input, variables['postcodes'])
+            except ValueError:
+                print("Sorry, that is an invalid postcode")
+                continue
+            if result == -1:
+                print("Could not find postcode")
+                continue
+            elif result == -2:
+                print("Multiple postcodes found")
+                continue
+            else:
+                print("Latitude: " + result[0] + " Longitude: " + result[1])
+                lat = result[0]
+                long = result[1]
+                postcodelatlong = []
+                postcodelatlong.append(float(lat))
+                postcodelatlong.append(float(long))
+                break
```

Comment:
will get marked down for using things like break and continue, I would suggest using a bool called valid or something